tests/unit_tests/test_wave_02.py

```diff
@@ -34,7 +34,12 @@ def test_get_no_matching_items_by_category():
 
     items = vendor.get_by_category("electronics")
 
-    raise Exception("Complete this test according to comments below.")
+    #raise Exception("Complete this test according to comments below.")
+    ###Assertions I added###
+    assert len(items) == 0
+    assert items == []
```

Comment:
We could assert the length of the inventory is the same as it was before the function call?

Reply:
Yeah, it would be good to check the length, and also check if the individual items in the vendor's inventory are still the same!

There's some things we need to consider about saving the inventory to a variable: 
- If we make a variable that points to the inventory list, if the inventory list changes, our variable will see those changes as well. 
- If we make a deep copy of the list, the Item objects in the copy will have different IDs than the Items in the original list and will compare to false. By default, Python will compare custom classes by their IDs when we use the `==` operator unless we override some other methods. 
- If we make a shallow copy though, the list IDs are different but the references inside the two lists are the same. If we compare the two lists of Items with `==` the lists will compare the ID of each Item at the corresponding indices, which should all be the same. If we could guarantee the order of the items in a vendor, then we could use a shallow list for comparison.

If we couldn't guarantee the order of the inventory, since we have the individual Item objects that should be in the list `item_a`, `item_b`, and `item_c` we could assert each item individually similar to the example the wave 1 tests' feedback or look into using the `all` function:
```python
assert all(item in vendor.inventory for item in [item_a, item_b, item_c])
```